migration-engine/connectors/sql-migration-connector/src/database_schema_differ.rs

```diff
@@ -208,4 +216,45 @@ impl<'a> DatabaseSchemaDiffer<'a> {
         }
         result
     }
+
+    /// An iterator over the tables that are present in both schemas. The yielded tuples should be interpreted as `(previous_table, next_table)`.
+    fn table_pairs(&self) -> impl Iterator<Item = (&Table, &Table)> {
+        self.previous.tables.iter().filter_map(move |previous_table| {
+            self.next
+                .tables
+                .iter()
+                .find(|next_table| next_table.name == previous_table.name)
+                .map(|next_table| (previous_table, next_table))
+        })
+    }
+
+    fn alter_indexes(&self) -> Vec<AlterIndex> {
+        self.table_pairs()
+            .flat_map(|(previous_table, next_table)| {
+                previous_table
+                    .indices
+                    .iter()
+                    .filter_map(move |previous_index| {
+                        next_table
+                            .indices
+                            .iter()
+                            .find(|next_index| {
+                                indexes_are_equivalent(previous_index, next_index)
+                                    && previous_index.name != next_index.name
+                            })
+                            .map(|renamed_index| (previous_index, renamed_index))
+                    })
+                    .map(move |(previous_index, renamed_index)| AlterIndex {
+                        index_name: previous_index.name.clone(),
+                        index_new_name: renamed_index.name.clone(),
+                        table: next_table.name.clone(),
+                    })
```

Comment:
The heavy iterator use here feels natural when I write it but I don't know if it's very readable with all that nesting. Maybe we should extract the code inside the `flat_map` to a separate method for clarity?